code/modules/wiremod/shell/brain_computer_interface.dm

```diff
@@ -10,6 +10,8 @@
 /obj/item/organ/internal/cyberimp/bci/Initialize(mapload)
 	. = ..()
 
+	create_reagents(15, TRANSPARENT | REFILLABLE) // This is used for the reagent injector component.
```

Comment:
The reagent_injector component itself should add this

Reply:
Then you could even transfer the reagents between different BCIs because when you take out the component it could move the reagents with it

Reply:
its intentionally made this way so that you can add multiple reagent injector components without any conflicts

Reply:
Multiple reagent injector components should be using the same reagents, and it should only exist if the reagent container is actually added. This doesn't conflict with what I want.

Reply:
> Multiple reagent injector components should be using the same reagents, and it should only exist if the reagent container is actually added. This doesn't conflict with what I want.

so like, wouldnt this still make you take out the component and whatnot? i just dont feel like making you reroute your circuitry just to refill it, ya know?

unless you feel like making me overload all the reagent procs to the BCI... in which case, why bother?

Reply:
No it would not because it would be creating the reagents on the BCI

Reply:
> 

All atoms have reagents datums, what's the point here? Most reagents datums are created at initialize, too. I also need some not-shit way of detecting if another component has already added it. Removing it isn't feasible, don't want to be deleting people's reagents.

With all this said, the current method is the most sane way of doing it.

Why does it even matter anyways? Do you care that much about a "It Contains:" message or why are you so adamant about it?

Reply:
A circuit that doesn't have a reagents component inside it should not have its own reagents container, I don't know why that's such a big deal

Reply:
> A circuit that doesn't have a reagents component inside it should not have its own reagents container, I don't know why that's such a big deal. It's silly UX otherwise, it would mean yes the description gives a useless line of text, but also that you would be able to inject reagents into something that it won't do anything with.

Give me a way of handling it without shitcode please, don't want removing the component to delete any reagents.

Reply:
Give the reagents component to the circuit component

When added to a BCI, transfer the reagents component if it does not already have one

If the component is taken out, and it is the last one that does not have reagents in it, transfer the reagents component to that